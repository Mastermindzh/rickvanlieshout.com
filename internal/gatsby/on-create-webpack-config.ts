import path from "path";

import { CreateWebpackConfigArgs } from "gatsby";
import { CompilerOptions } from "typescript";

import { compilerOptions } from "../../tsconfig.json";

const onCreateWebpackConfig = (
  (options: Pick<CompilerOptions, "paths">) =>
  ({ actions, getConfig }: CreateWebpackConfigArgs) => {
    // Keep existing TS path aliases in webpack
    actions.setWebpackConfig({
      resolve: {
        alias: Object.entries(options.paths || []).reduce(
          (aliases, [name, [target]]) => ({
            ...aliases,
            [name]: path.resolve(target),
          }),
          {},
        ),
      },
    });

    // Workaround: Gatsby's webpack ESLint plugin is incompatible with ESLint v9+ options
    // on some environments. Remove the plugin from the webpack pipeline during development
    // to prevent build failures like "Invalid Options: Unknown options: extensions, useEslintrc".
    // We still keep linting via npm scripts.
    try {
      const config = getConfig();
      if (config?.plugins?.length) {
        const beforeCount = config.plugins.length;
        config.plugins = config.plugins.filter((plugin: any) => {
          const name = plugin?.constructor?.name;
          return name !== "ESLintWebpackPlugin" && name !== "ESLintPlugin";
        });
        const afterCount = config.plugins.length;
        // Only replace when we've actually modified the plugins array
        if (afterCount !== beforeCount) {
          actions.replaceWebpackConfig(config);
        }
      }
    } catch {
      // noop â€“ if Gatsby changes internals, don't crash the build
    }
  }
)(compilerOptions);

export { onCreateWebpackConfig };
