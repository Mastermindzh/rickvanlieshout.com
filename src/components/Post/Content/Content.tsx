import React from "react";
import { useSiteMetadata } from "../../../hooks";
import { PostHeader } from "../../PostHeader/PostHeader";
import * as styles from "./Content.module.scss";

interface Props {
  title: string;
  body: string;
}

const Content: React.FC<Props> = ({ body, title }: Props) => {
  const { author } = useSiteMetadata();
  return (
    <>
      <PostHeader author={author} />
      <div className={styles.content}>
        <h1 className={styles.title}>{title}</h1>
        <div className={styles.body} dangerouslySetInnerHTML={{ __html: body }} />
      </div>
    </>
  );
};

export default Content;
