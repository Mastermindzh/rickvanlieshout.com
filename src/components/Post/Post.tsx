import React from "react";
import { Helmet } from "react-helmet";
import { useSiteMetadata } from "../../hooks";
import { Author } from "./Author";
import { Comments } from "./Comments";
import { Content } from "./Content";
import { Meta } from "./Meta";
import * as styles from "./Post.module.scss";
import { Tags } from "./Tags";
import type { Node } from "@/types";

interface Props {
  post: Node;
}

const Post: React.FC<Props> = ({ post }: Props) => {
  const { html } = post;
  const { tagSlugs, slug, readingTime } = post.fields;
  const { tags, title, date, disqusId, category } = post.frontmatter;
  const { url } = useSiteMetadata();

  return (
    <>
      <Helmet>
        <meta property="og:type" content="article" />
        <meta property="article:publisher" content={url} />
        <meta property="article:section" content={category} />
        <meta property="article:tag" content={tags ? tags[0] : ""} />
      </Helmet>

      <div className={styles.post}>
        <div className={styles.content}>
          <Content body={html} title={title} subTitle={readingTime?.text} />
        </div>

        <div className={styles.footer}>
          <Meta date={date} />
          {tags && tagSlugs && <Tags tags={tags} tagSlugs={tagSlugs} />}
          <Author />
        </div>

        <div className={`${styles.comments} hideInPrintView`}>
          <Comments disqusId={disqusId} postSlug={slug} postTitle={post.frontmatter.title} />
        </div>
      </div>
    </>
  );
};

export default Post;
