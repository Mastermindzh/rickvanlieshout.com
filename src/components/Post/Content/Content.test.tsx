import { Content } from "@/components/Post/Content";
import * as mocks from "@/mocks";
import { StaticQuery, useStaticQuery } from "gatsby";
import React from "react";
import renderer from "react-test-renderer";

const mockedStaticQuery = StaticQuery as jest.Mock;
const mockedUseStaticQuery = useStaticQuery as jest.Mock;

describe("Content", () => {
  beforeEach(() => {
    mockedStaticQuery.mockImplementationOnce(({ render }) => render(mocks.siteMetadata));
    mockedUseStaticQuery.mockReturnValue(mocks.siteMetadata);
  });
  it("renders correctly", () => {
    const props = {
      title: mocks.markdownRemark.frontmatter.title,
      body: mocks.markdownRemark.html,
    };

    const tree = renderer.create(<Content {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
