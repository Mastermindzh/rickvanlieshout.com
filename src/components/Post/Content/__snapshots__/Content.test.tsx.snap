// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders correctly 1`] = `
Array [
  <div>
    <span>
      <span>
        <a
          href="/"
        >
          Rick 
          <span>
            van Lieshout
          </span>
        </a>
      </span>
    </span>
    <span>
      <a
        href="/"
        style={
          Object {
            "lineHeight": "50px",
            "marginRight": "10px",
          }
        }
      >
        â†© All articles
      </a>
    </span>
  </div>,
  <div>
    <h1>
      Perfecting the Art of Perfection
    </h1>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>An Essay on Typography by Eric Gill takes the reader back to the year 1930. The year when a conflict between two worlds came to its term. The machines of the industrial world finally took over the handicrafts.</p>",
        }
      }
    />
  </div>,
]
`;
